conda_build/environ.py

```diff
@@ -47,6 +47,10 @@ def get_git_build_info(src_dir, git_url, expected_rev):
     env = os.environ.copy()
     d = {}
     git_dir = join(src_dir, '.git')
+    if not isinstance(git_dir, str):
+        # On Windows, subprocess env can't handle unicode.
+        git_dir = git_dir.encode(sys.getfilesystemencoding())
```

Comment:
I'm still a little nervous about this. I have seen encoding fixes like this before and ran into new problems because of them could we add a `or 'utf-8'` here.